### PR TITLE
Fix direct fresh capability normalization

### LINKED ISSUE #25652: CC: Inconsistent behavior in snippet compilation for fresh capabilities
## Compiler version

On 724e9c352b45453bd6a50857ed48191b4a84bb84

## Minimized code

Discovered on PR https://github.com/scala/scala3/pull/25612. Suddenly, the CI consistently fails on the following snippet during `scaladoc/generateReferenceDocumenation`:

```scala
package staticsitesnippet

import language.experimental.captureChecking

object Snippet:
  import caps.*
  class A
  type F[X] = (t: String) -> X
  type G[C^] = (t: String) -> A^{C}
  import caps.fresh

  val x: (s: String) -> F[A^{fresh}] = ???
  val y: (s: String) -> G[{fresh}] = ??? // <- error here
```

## Output

```scala
-- Error: /tmp/sbt_f9589512/_docs/reference/experimental/capture-checking/scoped-capabilities.md:358:7 
358 |val y: (s: String) -> G[{fresh}] = ???
    |       ^
    |Type argument scala.caps.CapSet^{fresh} does not conform to upper bound scala.caps.CapSet^ in subpart staticsitesnippet.Snippet.G[scala.caps.CapSet^{fresh}] of inferred type (s: String) -> staticsitesnippet.Snippet.G[scala.caps.CapSet^{fresh}]
    |
    |where:    ^     refers to a root capability in the type of type G
    |          fresh is a root capability associated with the result type of (s: String): staticsitesnippet.Snippet.G[scala.caps.CapSet^{fresh}]
```

## Expectation

It should compile.

### PR DESCRIPTION
**Pull request by bracevac**
Fixes #25652

The bug was discovered in https://github.com/scala/scala3/pull/25612.

The core issue was that `cc.Setup` stopped too early once a type contained direct `fresh`. That prevented a needed dealiasing step in cases like:

```scala
type G[cs^] = (t: String) -> A^{cs}
val y: (s: String) -> G[{fresh}] = ???
```

The fix is small: still try `mappedDealias`, and only keep the original type if that step makes no progress.

## How much have you relied on LLM-based tools in this contribution?

Minimally, let it point to the place where the problem originated.

## How was the solution tested?

This PR also adds a minimal positive regression test for the standalone `fresh` cases, and updates the existing `outer-fresh` negative checkfile to the cleaner diagnostics produced after the fix.

### CODE REVIEW

**Comment by bracevac** [comment]
Best PR number 😈

File: compiler/src/dotty/tools/dotc/cc/Setup.scala [hunk 1]
```diff
@@ -469,7 +469,8 @@ class Setup extends PreRecheck, SymTransformer, SetupAPI:
         case t1 @ CapturingType(parent, refs) =>
           t1.derivedCapturingType(stripImpliedCaptureSet(mapAndMaybeDealias(parent)), refs)
         case t1 =>
-          if t1.containsGlobalFreshDirectly then t1 else mappedDealias(t1)
+          val t2 = mappedDealias(t1)
+          if t1.containsGlobalFreshDirectly && (t2 eq t1) then t1 else t2
```
Comment (Linyxus) [inline compiler/src/dotty/tools/dotc/cc/Setup.scala:473]: This looks a bit suspicious to me. From my understanding we intentionally skip mapping t1 when it `containsGlobalFreshDirectly` since these freshes will be mapped to ResultCaps by a later mapping.

But I'm no way an expert here. So let's wait for @odersky 's opinion.

**Comment by bracevac** [comment]
> But it also looks to me that this example is not rejected "properly" right now but probably "by accident". So there are definitely things to fix and improve here.

IIRC It's actually a pos test case from when the fresh/any distinction was introduced.

**Comment by bracevac** [comment]
But digging further @Linyxus, I think you might be right here. The original test case is
```scala
import caps.fresh

class A

object Test:

  type F[X] = (t: String) -> X
  type G[cs^] = (t: String) -> A^{cs}

  val x: (s: String) -> F[A^{fresh}] = ???
  val _: (s: String) -> (t: String) -> A = x // error

  val y: (s: String) -> G[{fresh}] = ???
  val _: (s: String) -> (t: String) -> A = y // error

  val z: () -> F[A^{fresh}] = ???
  val _: () -> (t: String) -> A = z // error

  val z2: () -> G[{fresh}] = ???
  val _: () -> (t: String) -> A = z2 // error
```
and those errors _mask_ the error we see now. If we comment out everything but `G` and `y`, we see the error that shows up in the snippet compilation. 🙃
